data_structures/linked_list/__init__.py

```diff
@@ -21,8 +21,45 @@ def __init__(self) -> None:
         self.head: Node | None = None
         self.size = 0
 
-    def add(self, item: Any) -> None:
-        self.head = Node(item, self.head)
+    def add(self, item: Any, position: int = 0) -> None:
+        """
+        Add an item to the LinkedList at the specified position.
+        Default position is 0 (the head).
+
+        Args:
+            item (Any): The item to add to the LinkedList.
+            position (int, optional): The position at which to add the item.
+            Defaults to 0.
+
+        Raises:
+            ValueError: If the position is negative.
+
+        >>> linked_list = LinkedList()
+        >>> linked_list.add(1)
+        >>> linked_list.add(2)
+        >>> linked_list.add(3)
+        >>> linked_list.add(4, 2)
+        >>> print(linked_list)
+        3 --> 2 --> 4 --> 1
```

Comment:
Can you add doctests to test positions that are negative, out of bounds, and at the end of the linked list?